packages/sycamore-core/src/component.rs

```diff
@@ -176,3 +181,108 @@ impl<'a, G: GenericNode> Children<'a, G> {
         Self { f: Box::new(f) }
     }
 }
+
+/// The value of a passthrough attribute.
+/// The default for unknown attributes is [`AttributeValue::Str`] or [`AttributeValue::DynamicStr`]
+pub enum AttributeValue<'cx, G: GenericNode> {
+    /// A string literal value. Example: `attr:id = "test"`
+    Str(&'static str),
+    /// A dynamic string value from a variable. Example: `attr:id = id_signal`
+    DynamicStr(Box<dyn Display>),
```

Comment:
I still think it would be clearer to have a `Box<dyn FnMut() -> String>` instead. If you have a `ReadSignal`, you can easily create a closure with `|| signal.get().to_string()`

This way, it is immediately clear that there can be a reactive access inside the closure whereas the `dyn Display` is a lot more obscure.

Same goes for `DynamicBool` and `DynamicDangerouslySetInnerHtml`.

Reply:
I disagree but it's not a hard change I think so I'll just give it a try